Magento_Checkout/web/js/mixin/step-validation-mixin.js

```diff
@@ -0,0 +1,31 @@
+define([
+  'ko',
+  'Magento_Customer/js/model/customer',
+  'Magento_Checkout/js/model/step-navigator',
+], function (ko, customer, stepNavigator) {
+  return function (checkoutComponent) {
+    return checkoutComponent.extend({
+      defaults: {
+        isVisible: ko.observable(true),
+      },
+
+      initialize() {
+        this._super();
+        if (!customer.isLoggedIn()) {
+          this.isVisible(false);
+          stepNavigator.navigateTo('isLoggedCheck');
+        }
+      },
+
+      navigateToNextStep() {
+        if (customer.isLoggedIn()) {
+          this.isVisible(true);
+          stepNavigator.next();
+        } else {
+          this.isVisible(false);
+          stepNavigator.navigateTo('isLoggedCheck');
```

Comment:
I see you used the same code lines here and in the initialize method
```js
this.isVisible(false);
stepNavigator.navigateTo('isLoggedCheck');
```
I think it'd be better to move it into a new method not to duplicate
```js
initialize() {
  this._super();

  if (!customer.isLoggedIn()) {
    this.navigateToAuthStep();
  }
},

navigateToNextStep() {
  if (!customer.isLoggedIn()) {
    return this.navigateToAuthStep();
  }

  this.isVisible(true);
  stepNavigator.next();
}

navigateToAuthStep() {
    this.isVisible(false);
    stepNavigator.navigateTo('isLoggedCheck');
}
```